tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -3116,7 +3170,23 @@ ReductionCodegenInfo IrEmitterUnnested::ComputeReductionCodegenInfo(
       !IsUnrollingColumnReductionBeneficial(unnested_hlo, input_shape,
                                             reduction_dimensions.dimensions[2]);
 
-  if (!dilated_x && !reduction_dimensions.is_row_reduction) {
+  KernelMappingScheme::IndexingOrder indexing_order;
+  if (reduction_dimensions.is_row_reduction &&
+      // Only try to vectorize+coales memory access for row of even size.
```

Comment:
I tried adding an kPad node in the XLA graph. If you suggest a different padding, tell me. This have a few problems:
- This cause the fused reduction with exponential for softmax to be 2 fusion instead of one. 
   - The fusion input shape and reduction input shape aren't the same and the fusion optimizer reject that case.
   - I hacked a fix the fusion optimization, but this also cause problem in the generated code where the shapes isn't the right one.
- The final PTX code still ends up with some conditions in it. But the conditions break the vectorization for each some of the loads for all threads. So the code should be less optimized. I didn't benchmark if this make a difference, but the current partially optimized even case do not have all the speed up I was expecting if it was fully vectorized.

I have 2 ideas that could help odd row shapes. This could remove this need. But this would delay much this PR that is fully working and useful. Also it isn't even sure they will works. So I think it is useful to get this change in now.